tests/PatternAssertions/TextFeaturedMediaAssert.php

```diff
@@ -74,4 +77,57 @@ protected function assertVideoRatio(string $expected_ratio, string $selector, Cr
     }
   }
 
+  /**
+   * Asserts the link of the pattern.
+   *
+   * @param array $expected_link
+   *   Array with keys: 'label', 'path', 'icon'.
+   *   'icon' can be 'external' or 'corner-arrow'.
+   * @param \Symfony\Component\DomCrawler\Crawler $crawler
+   *   The DomCrawler where to check the element.
+   */
+  protected function assertLink(array $expected_link, Crawler $crawler): void {
+    $link_element = $crawler->filter('a.ecl-link.ecl-link--icon.ecl-link--icon-after.ecl-u-mt-m.ecl-u-mt-md-l.ecl-u-type-bold');
+    self::assertEquals($expected_link['path'], $link_element->attr('href'));
+
+    $label_element = $link_element->filter('span.ecl-link__label');
+    self::assertEquals($expected_link['label'], $label_element->text());
+
+    $svg = $link_element->filter('svg.ecl-icon.ecl-icon--s.ecl-icon--primary.ecl-link__icon use');
+    self::assertContains('icons.svg#ui--' . $expected_link['icon'], $svg->attr('xlink:href'));
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getPatternVariant(string $html): string {
+    // Variant can be clearly recognised if image or video and link exist only.
+    // In case if the link isn't provided then "left_simple", "right_simple"
+    // will be returned based on the position of media.
+    // In case if the image or video aren't provided then
+    // "left_simple", "left_featured" will be returned based on the link style.
+    // If image or video and link aren't provided then "left_simple" will be
+    // returned as default value.
+    $crawler = new Crawler($html);
+
+    $position_variant = 'left';
+    $media_position = $crawler->filter('.ecl-u-order-md-last');
```

Comment:
If there is no media this class is not present and the method will return a "right" type variant, but if there is no media we should return a "left" type variant right?

Reply:
You are right. I've added check that media wrapper exists to ensure that there is media.